### PR TITLE
Handling the new experimental daemon flag

### PR DESCRIPTION
**Pull request by mavenugo**
related to https://github.com/docker/docker/issues/29368

Signed-off-by: Madhu Venugopal <madhu@docker.com>

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -476,7 +476,7 @@ func (c *controller) ID() string {
 
 func (c *controller) BuiltinDrivers() []string {
 	drivers := []string{}
-	for _, i := range getInitializers() {
+	for _, i := range getInitializers(c.cfg.Daemon.Experimental) {
```
Comment (aboch) [inline controller.go]: Can we instead do something like:
```
initializers := getInitializers()
if c.cfg.Daemon.Experimental {
    initializers = append(initializers, additionalDrivers()...)
}
for _, i := range initializers {
...
```
and remove the call to `additionalDrivers()` from inside `getInitializers()`

Comment (mavenugo) [inline controller.go]: getInitializers is called from couple of places. so I would like to contain all the initializers in 1 place.

Comment (mavenugo) [inline controller.go]: But I can do your suggestion inside the getIntializers.

Comment (aboch) [inline controller.go]: > getInitializers is called from couple of places. so I would like to contain all the initializers in 1 place.

Ah, true.

> But I can do your suggestion inside the getIntializers.

Yes, thanks. That would work. I do not see a reason atm for sending the flag down to `additionalDrivers()`

Comment (mavenugo) [inline controller.go]: Done. PTAL.

**Comment by aboch** [comment]
LGTM

**Comment by sanimej** [comment]
LGTM

**Pull request merged**